src/Drupal/DKANExtension/Context/WorkflowContext.php

```diff
@@ -30,23 +35,27 @@ public function transitionModerationState($named_entity, $state, $date = null) {
       throw new \Exception("State '$state' is not available. All possible states are [$possible_states].");
     }
 
-    $current_user = $this->getCurrentUser();
+    $current_user = $user = ($workflow_user) ? user_load_by_name($workflow_user) : $this->getCurrentUser();
     if (!$current_user) {
       throw new \Exception("No user is logged in.");
     }
 
-    $my_revision = $node->workbench_moderation['my_revision'];
-    $next_states = workbench_moderation_states_next($my_revision->state, $current_user, $node);
-    if (empty($next_states)) {
-      $next_states = array();
-    }
-    if (!isset($next_states[$state_key])) {
-      $next_states = implode(", ", $next_states);
-      throw new \Exception("State '$possible_states[$state_key]' is not available to transition to. Transitions available to user '$current_user->name' are [$next_states]");
-    }
+    $my_revision = $node->workbench_moderation['my_revision'];;
+    $flipped_posible_states = array_flip($possible_states);
 
-    // This function actually updates the transition.
-    workbench_moderation_moderate($node, $state_key, $current_user->uid);
+    if($my_revision->state != $flipped_posible_states[$state]) {
```

Comment:
What's the purpose here? Won't this cause the step to not fail if the test step tries to transition to a current state?